stress_test/Main.cpp

```diff
@@ -1,28 +1,58 @@
 #include <chrono>
 #include <iostream>
+#include <thread>
 #include <vector>
 
 #include <gflags/gflags.h>
 
 #include "Mixer.h"
 
 DEFINE_int32(num_producers, 100, "number of producers to run");
+DEFINE_int32(num_threads, 1, "number of threads to run");
 
-int main(int argc, char **argv) {
+using std::make_shared;
+using std::shared_ptr;
+using std::vector;
+
+void createAndRunMixer(vector<shared_ptr<Producer>> producers, int me,
+                       vector<shared_ptr<ToFreeQueue>> toFreeQueues) {
+  Mixer m(producers, FLAGS_num_producers, me, toFreeQueues);
+  m.run();
+}
 
+int main(int argc, char **argv) {
   gflags::ParseCommandLineFlags(&argc, &argv, true);
 
-  vector<std::unique_ptr<Producer>> producers;
-  producers.push_back(std::move(std::make_unique<SimpleProducer>(8, 100000)));
-  producers.push_back(std::move(std::make_unique<VectorProducer>(
-      100000, std::chrono::duration<double>(1.0))));
+  // Initialize producers
+  vector<shared_ptr<Producer>> producers;
```

Comment:
In this iteration, the producers were basically immutable configuration objects, and each mixer ran the same set of producers, so it made sense to share producers. I've since changed this back to unique because these assumptions are no longer true.